bindings/go/blob/filesystem/file.go

```diff
@@ -109,6 +116,19 @@ func EnsurePathInWorkingDirectory(path, workingDirectory string) (_ string, err
 
 	_, err = os.OpenInRoot(workingDirectory, path)
 	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return "", fmt.Errorf("path %q does not exist in directory %q: %w %w", path, workingDirectory, ErrFileNotFound, err)
+		}
+		if errors.Is(err, os.ErrPermission) {
+			return "", fmt.Errorf("permission denied for path %q in directory %q: %v %w", path, workingDirectory, ErrPermissionDenied, err)
+		}
+		if errors.Is(err, os.ErrInvalid) {
+			return "", fmt.Errorf("invalid path %q in directory %q: %w %w", path, workingDirectory, ErrInvalidPath, err)
+		}
+		var pathError *fs.PathError
+		if errors.As(err, &pathError) {
+			return "", fmt.Errorf("path error for %q in directory %q: %w %w", path, workingDirectory, ErrInvalidPath, err)
+		}
 		return "", fmt.Errorf("failed to open path %q in root %q: %w", path, workingDirectory, err)
```

Comment:
Why do we need to wrap errors here? it feels like people could just check with Errors is for existing errors?